pkg/controllers/provisioning/scheduling/nodeclaimtemplate.go

```diff
@@ -57,7 +58,10 @@ func NewNodeClaimTemplate(nodePool *v1.NodePool) *NodeClaimTemplate {
 		v1.NodePoolHashAnnotationKey:        nodePool.Hash(),
 		v1.NodePoolHashVersionAnnotationKey: v1.NodePoolHashVersion,
 	})
-	nct.Labels = lo.Assign(nct.Labels, map[string]string{v1.NodePoolLabelKey: nodePool.Name})
+	nct.Labels = lo.Assign(nct.Labels, map[string]string{
+		v1.NodePoolLabelKey: nodePool.Name,
+		nodeclaimutils.NodeClassLabelKey(nodePool.Spec.Template.Spec.NodeClassRef): nodePool.Spec.Template.Spec.NodeClassRef.Name,
```

Comment:
Should this be a utils function or something on the apis package?

Reply:
Good call, this should probably be in the APIs package